### PR TITLE
feat: add data platform support

### PR DESCRIPTION
**Pull request by iblindu**
## Checklist

- [x] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [x] Tests added or updated
- [x] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [ ] Jira task updated

### CODE REVIEW

File: docs/data-sources/dataplatform_node_pools.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+subcategory: "Dataplatform"
+layout: "ionoscloud"
+page_title: "IonosCloud : ionoscloud_dataplatform_node_pools"
+sidebar_current: "docs-dataplatform_node_pools"
+description: |-
+Get information on a Dataplatform Node Pool list under a Dataplatform Cluster.
```
Comment (cristiGuranIonos) [inline docs/data-sources/dataplatform_node_pools.md:7]: todo: indentation

File: ionoscloud/data_source_dataplatform_cluster.go [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+							Type:      schema.TypeString,
+							Computed:  true,
+							Sensitive: true,
+						},
+						"current_context": {
+							Type:      schema.TypeString,
+							Computed:  true,
+							Sensitive: true,
+						},
+						"kind": {
```
Comment (adeatcu-ionos) [inline ionoscloud/data_source_dataplatform_cluster.go:84]: @cristiGuranIonos not sure about this: for other fields we have `Sensitive: true`. For `kind`, we don't. Is this intended or should be `kind` also be treated like sensitive?

File: ionoscloud/data_source_dataplatform_cluster.go [hunk 3]
```diff
@@ -0,0 +1,373 @@
...
+
+		log.Printf("[INFO] Using data source for Dataplatform Cluster by name with partial_match %t and name: %s", partialMatch, name)
+
+		if partialMatch {
+			clusters, _, err := client.ListClusters(ctx, name)
+			if err != nil {
+				diags := diag.FromErr(fmt.Errorf("an error occurred while fetching Dataplatform Clusters: %w", err))
+				return diags
+			}
+			results = *clusters.Items
```
Comment (adeatcu-ionos) [inline ionoscloud/data_source_dataplatform_cluster.go:223]: @cristiGuranIonos below we also make this check: `if clusters.Items != nil`, shouldn't we do the same for this one?

File: ionoscloud/data_source_dataplatform_cluster.go [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+		} else {
+			clusters, _, err := client.ListClusters(ctx, "")
+			if err != nil {
+				diags := diag.FromErr(fmt.Errorf("an error occurred while fetching Dataplatform Clusters: %w", err))
+				return diags
+			}
+			if clusters.Items != nil && len(*clusters.Items) > 0 {
+				for _, clusterItem := range *clusters.Items {
+					if clusterItem.Properties != nil && clusterItem.Properties.Name != nil && strings.EqualFold(*clusterItem.Properties.Name, name) {
+						tmpCluster, _, err := client.GetClusterById(ctx, *clusterItem.Id)
```
Comment (adeatcu-ionos) [inline ionoscloud/data_source_dataplatform_cluster.go:233]: @cristiGuranIonos first we make:

`clusters, _, err := client.ListClusters(ctx, "")`

Then we iterate through them make and make other requests for the clusters that match the name:

```
if clusterItem.Properties != nil && clusterItem.Properties.Name != nil && strings.EqualFold(*clusterItem.Properties.Name, name) {
						tmpCluster, _, err := client.GetClusterById(ctx, *clusterItem.Id)
```
Is this necessary? Can't we just iterate through previously retrieved `clusters` and build the list of results based on the name? Why do we need to make additional requests?

More than that, after building the list of `results`, we check if the length is greater than 1 and, if it is, we raise an error:
```
} else if len(results) > 1 {
			return diag.FromErr(fmt.Errorf("more than one Dataplatform Cluster found with the specified criteria name = %s", name))
}
```
I think that a better approach would be to raise an error in a loop, as soon as we observe that the length is  greater than 1. Basically we are making some useless requests because if the length is greater than 1, we still raise an error.

**Review by adeatcu-ionos — CHANGES_REQUESTED** [review]

File: ionoscloud/data_source_dataplatform_cluster.go [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+
+func dataSourceDataplatformCluster() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceDataplatformClusterRead,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:         schema.TypeString,
+				Description:  "The id of your cluster.",
+				Optional:     true,
+				ValidateFunc: validation.All(validation.StringMatch(regexp.MustCompile("^[A-Za-z0-9][-A-Za-z0-9_.]*[A-Za-z0-9]$"), "")),
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/data_source_dataplatform_cluster.go:27]: will change to ValidateDiagFunc

File: ionoscloud/data_source_dataplatform_cluster.go (hunk 2 above)
Comment (cristiGuranIonos) [reply ionoscloud/data_source_dataplatform_cluster.go:84]: it's only the type of k8s object

File: docs/data-sources/dataplatform_cluster.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+* `datacenter_id` - The UUID of the virtual data center (VDC) the cluster is provisioned.
+* `name` - The name of your cluster.
+* `data_platform_version` - The version of the Data Platform.
+* `maintenance_window` - Starting time of a weekly 4 hour-long window, during which maintenance might occur in hh:mm:ss format
+  * `time` - Time at which the maintenance should start. 
+  * `day_of_the_week`
+* `kube_config` - Kubernetes configuration
+* `config` - structured kubernetes config consisting of a list with 1 item with the following fields:
+  * api_version - Kubernetes API Version
+  * kind - "Config"
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_cluster.md:69]: same as our k8s config docs. Might be a good idea at some point

File: docs/data-sources/dataplatform_cluster.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+description: |-
+  Get information on a Dataplatform Cluster.
+---
+
+# ionoscloud_dataplatform_cluster
+
+⚠️ **Note:** Data Platform is currently in the Early Access (EA) phase.
+We recommend keeping usage and testing to non-production critical applications.
+Please contact your sales representative or support for more information.
+Manages a **Dataplatform Node Pool**.
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_cluster.md]: done

File: docs/data-sources/dataplatform_cluster.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+data "ionoscloud_dataplatform_cluster" "example" {
+  name	= "_Example"
+  partial_match = true
+}
+```
+
+## Argument Reference
+
+* `id` - (Optional) ID of the cluster you want to search for.
+* `name` - (Optional) Name or an existing cluster that you want to search for. Search by name is case-insensitive. The whole resource name is required if `partial_match` parameter is not set to true..
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_cluster.md]: done

[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_cluster.md]: done

File: docs/data-sources/dataplatform_cluster.md [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+* `partial_match` - (Optional) Whether partial matching is allowed or not when using name argument. Default value is false.
+
+Either `id` or `display_name` must be provided. If none, or both are provided, the datasource will return an error.
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_cluster.md]: done

File: ionoscloud/data_source_dataplatform_cluster.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/data_source_dataplatform_cluster.go]: done

File: ionoscloud/data_source_dataplatform_cluster.go (hunk 4 above)
Comment (cristiGuranIonos) [reply ionoscloud/data_source_dataplatform_cluster.go]: done

File: ionoscloud/resource_dataplatform_nodepool_test.go [hunk 10]
```diff
@@ -0,0 +1,378 @@
+//go:build all || dataplatform
```
Comment (adeatcu-ionos) [inline ionoscloud/resource_dataplatform_nodepool_test.go:1]: @cristiGuranIonos same observations I made for the cluster tests are valid here. Maybe we can make another PR in which we will add more tests for these.

File: services/dataplatform/client.go [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+	sdkClient dataplatform.APIClient
+}
+
+func NewClient(username, password, token, url, version, terraformVersion string) *Client {
+	newConfigDataplatform := dataplatform.NewConfiguration(username, password, token, url)
+
+	if os.Getenv("IONOS_DEBUG") != "" {
+		newConfigDataplatform.Debug = true
+	}
+	newConfigDataplatform.MaxRetries = 999
```
Comment (adeatcu-ionos) [inline services/dataplatform/client.go:25]: @cristiGuranIonos I noticed that these values are the same for every client, we should store them in a constant somewhere and reuse that, let's do this in another PR.

File: services/dataplatform/nodepool.go [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+}
+func (c *Client) IsNodePoolReady(ctx context.Context, d *schema.ResourceData) (bool, error) {
+	clusterId, ok := d.GetOk("cluster_id")
+	if !ok {
+		return false, fmt.Errorf("could not get cluster_id from schema")
+	}
+	clusterIdStr := clusterId.(string)
+	subjectNodePool, _, err := c.GetNodePool(ctx, clusterIdStr, d.Id())
+	if err != nil {
+		return true, fmt.Errorf("checking Dataplatform Node Pool status: %w", err)
```
Comment (adeatcu-ionos) [inline services/dataplatform/nodepool.go:68]: @cristiGuranIonos `error checking ..`. Also, shouldn't we return `false` here since the node pool is not ready?

File: services/dataplatform/cluster.go [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+	createRequest := setCreateClusterRequestProperties(d)
+	clusterResponse, apiResponse, err := c.sdkClient.DataPlatformClusterApi.CreateCluster(ctx).CreateClusterRequest(*createRequest).Execute()
+	apiResponse.LogInfo()
+	if err != nil {
+		return "", apiResponse, err
+	}
+	return *clusterResponse.Id, apiResponse, err
+}
+
+func (c *Client) UpdateCLuster(ctx context.Context, id string, d *schema.ResourceData) (utils.ApiResponseInfo, error) {
```
Comment (adeatcu-ionos) [inline services/dataplatform/cluster.go:68]: @cristiGuranIonos typo here, it should be `UpdateCluster`.

File: services/dataplatform/nodepool.go [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+			nodePools = append(nodePools, nodePoolEntry)
+		}
+
+		if nodePools == nil || len(nodePools) == 0 {
+			return diag.FromErr(fmt.Errorf("no nodepools found for criteria, please check your filter configuration"))
+		}
+
+		err := d.Set("node_pools", nodePools)
+		if err != nil {
+			diags := diag.FromErr(fmt.Errorf("while setting node_pools: %w", err))
```
Comment (adeatcu-ionos) [inline services/dataplatform/nodepool.go:312]: @cristiGuranIonos `error while ...` or `error occurred while ...`.

File: services/dbaas/cluster.go [hunk 15]
```diff
@@ -53,7 +53,6 @@ func (c *MongoClient) GetTemplates(ctx context.Context) (mongo.TemplateList, *mo
 
 func (c *PsqlClient) CreateCluster(ctx context.Context, cluster psql.CreateClusterRequest) (psql.ClusterResponse, *psql.APIResponse, error) {
 	clusterResponse, apiResponse, err := c.sdkClient.ClustersApi.ClustersPost(ctx).CreateClusterRequest(cluster).Execute()
-	apiResponse.LogInfo()
```
Comment (adeatcu-ionos) [inline services/dbaas/cluster.go:56]: @cristiGuranIonos why did we remove this?

**Review by adeatcu-ionos — CHANGES_REQUESTED** [review]

File: docs/data-sources/dataplatform_cluster.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply docs/data-sources/dataplatform_cluster.md:68]: @cristiGuranIonos please note everything that we decide to do in the future so we don't forget about them, this is the first one for this PR but in the last review I also wrote about 2-3 things that we will need to do in the future, so please note them somewhere, thank you!

File: services/dataplatform/nodepool.go (hunk 14 above)
Comment (cristiGuranIonos) [reply services/dataplatform/nodepool.go:312]: no need, wrapping the error will make it clear what happens. adding error is not useful here.

File: docs/data-sources/dataplatform_cluster.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_cluster.md:68]: jira issue

File: ionoscloud/data_source_dataplatform_cluster.go [hunk 16]
```diff
@@ -0,0 +1,374 @@
...
+	clustersList := make([]map[string]interface{}, len(kubeConfig.Clusters))
+	for i, cluster := range kubeConfig.Clusters {
+
+		/* decode ca */
+		decodedCrt := make([]byte, base64.StdEncoding.DecodedLen(len(cluster.Cluster.CaData)))
+		if _, err := base64.StdEncoding.Decode(decodedCrt, []byte(cluster.Cluster.CaData)); err != nil {
+			return err
+		}
+
+		if len(caCrt) == 0 {
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/data_source_dataplatform_cluster.go:313]: it's a for loop. it gets initialised only once

File: services/dataplatform/client.go (hunk 11 above)
Comment (cristiGuranIonos) [reply services/dataplatform/client.go]: we hvae the constants

File: ionoscloud/data_source_dataplatform_cluster.go (hunk 16 above)
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply ionoscloud/data_source_dataplatform_cluster.go:313]: @cristiGuranIonos oh, yes, my bad.

File: services/dataplatform/nodepool.go [hunk 17]
```diff
@@ -0,0 +1,317 @@
...
+	return apiResponse.HttpNotFound(), err
+}
+
+func (c *Client) ListNodePools(ctx context.Context, clusterId string) (dataplatform.NodePoolListResponseData, *dataplatform.APIResponse, error) {
+	nodePool, apiResponse, err := c.sdkClient.DataPlatformNodePoolApi.GetClusterNodepools(ctx, clusterId).Execute()
+	apiResponse.LogInfo()
+	return nodePool, apiResponse, err
+}
+
+func (c *Client) CreateNodePool(ctx context.Context, clusterId string, d *schema.ResourceData) (dataplatform.NodePoolResponseData, *dataplatform.APIResponse, error) {
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply services/dataplatform/nodepool.go:41]: We only use the `id` from the response though. Returning an `id` decouples the return of the function from the sdk specific things. This function is actually linked and even returns the apiresponse even though we don't use it. This function would maybe be better rewritten, but not in this PR.

File: docs/data-sources/dataplatform_node_pools.md [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+  cluster_id  = <cluster_id>
+}
+```
+
+### By Name
+
+```hcl
+data "ionoscloud_dataplatform_node_pools" "example" {
+  cluster_id    = <cluster_id>
+  name      	= "Dataplatform_Node_Pool_Example"
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/data-sources/dataplatform_node_pools.md:32]: It's in the description `The **Dataplatform Node Pools Data Source** can be used to search for and return a list of existing Dataplatform Node Pools under a Dataplatform Cluster.`. Also the fact that it's plurat - nodepools

File: services/dataplatform/nodepool.go (hunk 17 above)
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply services/dataplatform/nodepool.go:41]: @cristiGuranIonos I will create a Jira issue in order to discuss this.

File: ionoscloud/resource_dataplatform_cluster.go [hunk 19]
```diff
@@ -0,0 +1,184 @@
...
+	}
+}
+
+func resourceDataplatformClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(SdkBundle).DataplatformClient
+
+	id, _, err := client.CreateCluster(ctx, d)
+
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("while creating a Dataplatform Cluster: %w", err))
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go:78]: no need to add 'error' every time you wrap an existing error

File: ionoscloud/resource_dataplatform_cluster.go [hunk 20]
```diff
@@ -0,0 +1,184 @@
...
+		return diags
+	}
+
+	d.SetId(id)
+
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("while dataplaform cluster waiting to be ready: %w", err))
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go:86]: no need to add 'error' every time you wrap an existing error

File: ionoscloud/resource_dataplatform_cluster.go [hunk 21]
```diff
@@ -0,0 +1,184 @@
...
+func resourceDataplatformClusterRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	client := meta.(SdkBundle).DataplatformClient
+
+	clusterId := d.Id()
+	dataplatformCluster, apiResponse, err := client.GetClusterById(ctx, clusterId)
+
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go:102]: depends what type of function it is. read functions just need ot set the id to ""

File: ionoscloud/resource_dataplatform_cluster.go (hunk 20 above)
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply ionoscloud/resource_dataplatform_cluster.go:86]: @cristiGuranIonos yes yes you can ignore these comments if they are not required.

File: ionoscloud/resource_dataplatform_cluster.go (hunk 21 above)
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go:102]: we can add a debug message, but that's it

File: services/dataplatform/cluster.go [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+	return strings.EqualFold(*cluster.Metadata.State, utils.Available), nil
+}
+
+func (c *Client) GetClusterById(ctx context.Context, id string) (dataplatform.ClusterResponseData, *dataplatform.APIResponse, error) {
+	cluster, apiResponse, err := c.sdkClient.DataPlatformClusterApi.GetCluster(ctx, id).Execute()
+	apiResponse.LogInfo()
+	return cluster, apiResponse, err
+}
+
+// IsClusterDeleted - returns apiResponse to check if resource still exists. To be used with WaitForResourceToBeDeleted
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply services/dataplatform/cluster.go:33]: It's tricky, code should be commented only when it's not clear what it does. I added the comment here because I want to show how the function is used with WaitForResourceToBeDeleted. Even here, the comment is no longer in sync with what the function does.

File: services/dataplatform/cluster.go [hunk 23]
```diff
@@ -0,0 +1,192 @@
...
+	return apiResponse, err
+}
+
+func setCreateClusterRequestProperties(d *schema.ResourceData) *dataplatform.CreateClusterRequest {
+
+	dataplatformCluster := dataplatform.CreateClusterRequest{
+		Properties: &dataplatform.CreateClusterProperties{},
+	}
+
+	if nameValue, ok := d.GetOk("name"); ok {
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply services/dataplatform/cluster.go:87]: probably not, but it does not hur

File: ionoscloud/resource_dataplatform_cluster_test.go [hunk 24]
```diff
@@ -0,0 +1,220 @@
...
+}
+
+resource ` + DataplatformClusterResource + ` ` + DataplatformClusterTestResource + ` {
+  datacenter_id   		=  ` + DatacenterResource + `.datacenter_example.id
+  name 					= "` + UpdatedResources + `"
+  maintenance_window {
+    day_of_the_week  	= "Saturday"
+    time				= "10:00:00"
+  }
+  data_platform_version	= ` + DataPlatformVersion + `
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster_test.go]: can't be updated

File: docs/resources/dataplatform_cluster.md [hunk 25]
```diff
@@ -0,0 +1,54 @@
...
+* `maintenance_window` - (Optional)[string] Starting time of a weekly 4 hour-long window, during which maintenance might occur in hh:mm:ss format
+  * `time` - (Required)[string] Time at which the maintenance should start. Must conform to the 'HH:MM:SS' 24-hour format. This pattern matches the "HH:MM:SS 24-hour format with leading 0" format. For more information take a look at [this link](https://stackoverflow.com/questions/7536755/regular-expression-for-matching-hhmm-time-format).
+  * `day_of_the_week` - (Required)[string] Must be set with one the values `Monday`, `Tuesday`, `Wednesday`, `Thursday`, `Friday`, `Saturday` or `Sunday`.
+
+## Import
+
+Resource Dataplatform Cluster can be imported using the `cluster_id`, e.g.
+
+```shell
+terraform import ionoscloud_dataplatform_cluster.mycluser {cluster uuid}
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply docs/resources/dataplatform_cluster.md:52]: basically this can be whatever you want. It doesn't matter.

File: ionoscloud/resource_dataplatform_cluster_test.go [hunk 26]
```diff
@@ -0,0 +1,220 @@
+//go:build all || dataplatform
```
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster_test.go:1]: good point. separate pr though. will create issue

File: ionoscloud/resource_dataplatform_cluster.go (hunk 21 above)
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply ionoscloud/resource_dataplatform_cluster.go:102]: @cristiGuranIonos we have some read functions, for which we are doing this (check `nic` resource, for example) -> `we can add a debug message, but that's it`, I suggest doing the same here, but if you consider this isn't necessary we can skip it.

File: services/dataplatform/cluster.go (hunk 22 above)
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply services/dataplatform/cluster.go]: @cristiGuranIonos `It's tricky, code should be commented only when it's not clear what it does` I agree, sorry, I didn't actually read the code I only looked if we have function docs or not.

`Even here, the comment is no longer in sync with what the function does.` -- Then maybe we should sync it again by updating the comment?

File: ionoscloud/resource_dataplatform_nodepool.go [hunk 27]
```diff
@@ -0,0 +1,252 @@
...
+	dataplatformNodePoolResponse, _, err := client.CreateNodePool(ctx, clusterId, d)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("an error occured while creating a Dataplatform NodePool: %w", err))
+		return diags
+	}
+
+	d.SetId(*dataplatformNodePoolResponse.Id)
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsNodePoolReady)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf(" while dataplaform nodepool waiting to be ready: %w", err))
```
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply ionoscloud/resource_dataplatform_nodepool.go]: @cristiGuranIonos isn't this the same error wrap case in which we don't need to add `error` anymore? As it happened for the other useless comments that I left that were about the same thing? Am I missing something? Also, rereading this message I think it should be: `while waiting for dataplatform node pool to be ready`, not sure if how it is written now is correct in English.

File: ionoscloud/resource_dataplatform_nodepool.go [hunk 28]
```diff
@@ -0,0 +1,252 @@
...
+		ReadContext:   resourceDataplatformNodePoolRead,
+		UpdateContext: resourceDataplatformNodePoolUpdate,
+		DeleteContext: resourceDataplatformNodePoolDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceDataplatformNodePoolImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:             schema.TypeString,
+				Description:      "The name of your node pool. Must be 63 characters or less and must be empty or begin and end with an alphanumeric character ([a-z0-9A-Z]) with dashes (-), underscores (_), dots (.), and alphanumerics between.",
```
[the remark this answers is not in the document]
Comment (adeatcu-ionos) [reply ionoscloud/resource_dataplatform_nodepool.go]: @cristiGuranIonos this one isn't clear yet in my opinion, what bothers me is that we don't have any comma or dot between `begin and end with an alphanumeric character ([a-z0-9A-Z])` and `with dashes (-)`, we should separate them as I suggested or in another way. This applies for each comment related to this.

**Comment by adeatcu-ionos** [comment]
@cristiGuranIonos I think this should be ready tomorrow. Good job!! I know it's difficult to pick up such a big PR that was started by someone else.

File: ionoscloud/resource_dataplatform_cluster.go (hunk 21 above)
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_cluster.go:102]: ah, that's a very specific nic error case. will add  a log here just to be sure

File: ionoscloud/resource_dataplatform_nodepool.go (hunk 27 above)
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply ionoscloud/resource_dataplatform_nodepool.go]: Basically realized there are too many to change and no worth it. So we keep consistency even if the "an error occurred" does not add anything of value to the error message.

File: services/dataplatform/cluster.go (hunk 22 above)
[the remark this answers is not in the document]
Comment (cristiGuranIonos) [reply services/dataplatform/cluster.go]: yes, already done

**Review by adeatcu-ionos — APPROVED** [review]
